typegen/src/utils.rs

```diff
@@ -140,52 +156,58 @@ fn types_equal_inner(
         }
     }
 
-    let a_ty = types.resolve(a).expect("type a should exist in registry");
-    let b_ty = types.resolve(b).expect("type b should exist in registry");
-
     // Paths differ; types won't be equal then!
     if a_ty.path.segments != b_ty.path.segments {
         return false;
     }
 
-    // Names of type params don't line up, so different then!
-    if a_ty
-        .type_params
-        .iter()
-        .zip(&b_ty.type_params)
-        .any(|(a, b)| a.name != b.name)
-    {
-        return false;
-    }
+    #[rustfmt::skip]
+    let mut compare_fields = |
+        a: &Field<PortableForm>,
+        a_params: &GenericsList,
+        b: &Field<PortableForm>,
+        b_params: &GenericsList|
+     -> bool {
+        if a.name != b.name {
+            return false;
+        }
 
-    // We'll lazily extend our type params only if the shapes match.
-    let calc_params = || {
-        let a_params = a_parent_params.extend(&a_ty.type_params);
-        let b_params = b_parent_params.extend(&b_ty.type_params);
-        (a_params, b_params)
+        // Vec<T>, Option<T>, etc and #[scale_info(skip_type_params(T))]
```

Comment:
The variable is name is good but I had some troubles to understand the comment

```suggestion
        // The type is wrapped in another type such as `Vec<T>` or 
        // marked as skipped with `#[scale_info(skip_type_params(T))]`
```